File: internal/sem/testdata/langmatrix/javascript/ledger.js
export class Ledger {
  constructor() {
    this.total = 0;
  }

  add(amount) {
    return this.total + amount;
  }
}

export function ledgerDouble(amount) {
  const ledger = new Ledger();
  return ledger.add(amount) * 2;
}

File: internal/sem/testdata/langmatrix/scala/Ledger.scala
package fixtures

class Ledger(var total: Int = 0) {
  def add(amount: Int): Int = total + amount
}

object LedgerApp {
  def ledgerDouble(amount: Int): Int = {
    val ledger = new Ledger()
    ledger.add(amount) * 2
  }
}
